src/main/java/org/prebid/server/bidder/triplelift/TripleliftBidder.java

```diff
@@ -0,0 +1,152 @@
+        this.endpointUrl = HttpUtil.validateUrl(Objects.requireNonNull(endpointUrl));
+    }
+
+    @Override
+    public final Result<List<HttpRequest<BidRequest>>> makeHttpRequests(BidRequest bidRequest) {
+        final List<BidderError> errors = new ArrayList<>();
+        final List<Imp> validImps = new ArrayList<>();
+        for (Imp imp : bidRequest.getImp()) {
+            try {
+                validImps.add(modifyImp(imp));
+            } catch (PreBidException e) {
+                errors.add(BidderError.badInput(e.getMessage()));
+            }
+        }
+
+        if (validImps.isEmpty()) {
+            errors.add(BidderError.badInput("No valid impressions for triplelift"));
+            return Result.of(Collections.emptyList(), errors);
+        }
+
+        final BidRequest updatedRequest = bidRequest.toBuilder()
+                .imp(validImps)
+                .build();
+
+        return Result.of(Collections.singletonList(
+                HttpRequest.<BidRequest>builder()
+                        .method(HttpMethod.POST)
+                        .uri(endpointUrl)
+                        .body(Json.encode(updatedRequest))
+                        .headers(HttpUtil.headers())
+                        .payload(updatedRequest)
+                        .build()),
+                errors);
+    }
+
+    private static Imp modifyImp(Imp imp) throws PreBidException {
+        final ExtImpTriplelift impExt;
+        try {
+            impExt = Json.mapper.<ExtPrebid<?, ExtImpTriplelift>>convertValue(imp.getExt(),
+                    TRIPLELIFT_EXT_TYPE_REFERENCE).getBidder();
+        } catch (IllegalArgumentException e) {
+            throw new PreBidException(e.getMessage(), e);
+        }
+
+        if (imp.getBanner() == null && imp.getVideo() == null) {
+            throw new PreBidException("neither Banner nor Video object specified");
+        }
+
+        final Imp.ImpBuilder impBuilder = imp.toBuilder().tagid(impExt.getInventoryCode());
+        if (impExt.getFloor() != null) {
+            impBuilder.bidfloor(impExt.getFloor());
+        }
+
+        return impBuilder.build();
+    }
+
+    @Override
+    public Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
+        final BidResponse bidResponse;
+        try {
+            bidResponse = Json.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
+        } catch (DecodeException | PreBidException e) {
+            return Result.emptyWithError(BidderError.badServerResponse(e.getMessage()));
+        }
+
+        if (bidResponse == null || bidResponse.getSeatbid() == null) {
+            return Result.of(Collections.emptyList(), Collections.emptyList());
+        }
+
+        final List<BidderError> errors = new ArrayList<>();
+        final List<BidderBid> bidderBids = new ArrayList<>();
+        for (SeatBid seatBid : bidResponse.getSeatbid()) {
+            for (Bid bid : seatBid.getBid()) {
+                try {
```

Comment:
We don't need to open a try block here, but before the code that can throw the expected exception. Please move it before `treeToValue` call.